flask_rebar/swagger_generation/swagger_generator_v3.py

```diff
@@ -0,0 +1,322 @@
+from flask_rebar.swagger_generation.swagger_objects import Server
+from flask_rebar.authenticators import USE_DEFAULT
+from flask_rebar.swagger_generation.generator_utils import (
+    format_path_for_swagger,
+    verify_parameters_are_the_same,
+    get_response_description,
+    get_unique_schema_definitions,
+    recursively_convert_dict_to_ordered_dict,
+    get_ref_schema,
+)
+from flask_rebar.swagger_generation.marshmallow_to_swagger import (
+    get_swagger_title,
+    get_schema_fields,
+)
+from flask_rebar.validation import Error
+
+
+class SwaggerV3Generator(SwaggerGenerator):
+    """Generates a v3.0.2 Swagger specification from a Rebar object.
+
+    Not all things are retrievable from the Rebar object, so this
+    guy also needs some additional information to complete the job.
+
+    :param ConverterRegistry query_string_converter_registry:
+    :param ConverterRegistry request_body_converter_registry:
+    :param ConverterRegistry headers_converter_registry:
+    :param ConverterRegistry response_converter_registry:
+        ConverterRegistrys that will be used to convert Marshmallow schemas
+        to the corresponding types of swagger objects. These default to the
+        global registries.
+
+    :param Sequence[Tag] tags:
+        A list of tags used by the specification with additional metadata.
+    :param Sequence[Server] servers:
+        A list of Server Objects to set as the server metadata for the specification.
+    """
+
+    def __init__(
+        self,
+        version="1.0.0",
+        title="My API",
+        description="",
+        query_string_converter_registry=None,
+        request_body_converter_registry=None,
+        headers_converter_registry=None,
+        response_converter_registry=None,
+        tags=None,
+        servers=None,
+        default_response_schema=Error(),
+    ):
+        super(SwaggerV3Generator, self).__init__(
+            openapi_major_version=3,
+            version=version,
+            title=title,
+            description=description,
+            default_response_schema=default_response_schema,
+            query_string_converter_registry=query_string_converter_registry,
+            request_body_converter_registry=request_body_converter_registry,
+            headers_converter_registry=headers_converter_registry,
+            response_converter_registry=response_converter_registry,
+        )
+        self.tags = tags
+        self.servers = servers
+        self._ref_base = "#/components/schemas"
+
+    def generate(
+        self,
+        registry,
+        host=None,
+        schemes=None,
+        consumes=None,
+        produces=None,
+        sort_keys=True,
+    ):
+        """Generates a Swagger specification from a Rebar instance.
+
+        :param flask_rebar.rebar.HandlerRegistry registry:
+        :param str host: Adds this host as a Server Object for the service
+        :param schemes: IGNORED, but kept for compatibility with SwaggerV2Generator
```

Comment:
Do we need compatibility? If I change my app from `SwaggerV2Generator.generate()` to `SwaggerV3Generator.generate()`, I would expect that the params might change.

Reply:
I'd agree (that we probably don't need to include these), particularly given the notes below about consumes/produces.  I could see an ignored param being confusing for somebody that's starting with V3

Reply:
Assuming I haven't abused GitHub's forking and PR too badly for it to do what I intended, pushed a PR to Barak's fork which I think should address this: https://github.com/barakalon/flask-rebar/pull/1

Reply:
This should be addressed now